dlt/destinations/dataset/relation.py

```diff
@@ -205,29 +207,47 @@ def _query(self) -> Any:
 
         dataset_schema = self._dataset.schema
 
-        table_name = self.sql_client.make_qualified_table_name(
-            dataset_schema.naming.normalize_path(self._table_name)
+        table_name = dataset_schema.naming.normalize_tables_path(self._table_name)
```

Comment:
There is a change now on devel that runs the result of the query method through a qualifying and normalization step, have a look at what is there now, and you will see that you will not have to do any destination specific stuff or add database and catalog information here.